benches/lib.rs

```diff
@@ -733,44 +730,10 @@ impl BenchShieldedCtx {
         source: TransferSource,
         target: TransferTarget,
     ) -> Tx {
-        let mock_args = TxArgs {
-            dry_run: false,
-            dry_run_wrapper: false,
-            dump_tx: false,
-            force: false,
-            broadcast_only: false,
-            ledger_address: (),
-            initialized_account_alias: None,
-            fee_amount: None,
-            fee_token: address::nam(),
-            fee_unshield: None,
-            gas_limit: GasLimit::from(u64::MAX),
-            expiration: None,
-            disposable_signing_key: false,
-            signing_keys: vec![defaults::albert_keypair()],
-            signatures: vec![],
-            wallet_alias_force: true,
-            chain_id: None,
-            tx_reveal_code_path: TX_REVEAL_PK_WASM.into(),
-            verification_key: None,
-            password: None,
-            wrapper_fee_payer: None,
-            output_folder: None,
-        };
-
-        let args = TxTransfer {
-            tx: mock_args,
-            source: source.clone(),
-            target: target.clone(),
-            token: address::nam(),
-            amount: InputAmount::Validated(DenominatedAmount {
-                amount,
-                denom: 0.into(),
-            }),
-            native_token: self.shell.wl_storage.storage.native_token.clone(),
-            tx_code_path: TX_TRANSFER_WASM.into(),
+        let denominated_amount = DenominatedAmount {
+            amount,
+            denom: 0.into(),
```

Comment:
we default to the minimum denomination for all ibc amounts? this could be improved, since it steers away from the default behavior of the namada cli in all other instances (i.e. we always validate denominations with rpc calls)

if possible, we would query the IBC base denomination, then, using the obtained token address, query Namada's denomination in storage, eventually defaulting to 0, like we do for non-usable erc20 tokens

Reply:
I guess that this is just for a bench.